snakemake/satija/src/convert_h5ad_to_arrow.py

```diff
@@ -0,0 +1,42 @@
+import argparse
+
+from anndata import read_h5ad
+import pyarrow as pa
+import pandas as pd
+
+
+def h5ad_to_arrow(h5ad_file, arrow_file):
+    ann_data = read_h5ad(h5ad_file)
+    umap = ann_data.obsm['X_umap'].transpose()
+    leiden = ann_data.obs['leiden'].to_numpy().astype('uint8')
+    index = ann_data.obs.index
+
+    df = pd.DataFrame(
+        data={'umap_x': umap[0], 'umap_y': umap[1], 'leiden': leiden},
```

Comment:
I'm guessing the `umap` coordinates are `float64`. If you think we need that resolution, then I'd keep, put otherwise I'd cast down to `float32` to reduce the size of the binary.